dracut.sh

```diff
@@ -2118,16 +2080,28 @@ fi
 
 if dracut_module_included "squash"; then
     dinfo "*** Squashing the files inside the initramfs ***"
+    declare squash_compress_arg
+    if [[ $compress_l ]]; then
+        if ! mksquashfs /dev/null "$DRACUT_TMPDIR"/.squash-test.img -no-progress -comp $compress_l &>/dev/null; then
+            dwarn "mksquashfs doesn't support compressor '$compress_l', failing back to default compressor."
+        else
+            squash_compress_arg="-comp $compress_l"
+        fi
+    fi
```

Comment:
I think you should use `$compress` rather then `$compress_l`, which is only the CLI parameter, if set and not the setting from any configuration file.

```suggestion
    if [[ $compress ]]; then
        if ! mksquashfs /dev/null "$DRACUT_TMPDIR"/.squash-test.img -no-progress -comp $compress &>/dev/null; then
            dwarn "mksquashfs doesn't support compressor '$compress', failing back to default compressor."
        else
            squash_compress_arg="-comp $compress"
        fi
    fi
```

Reply:
I think I'll have to rework this part, $compress will be override by compressor detection code at line 924, and will contain invalid params for mksquash.